### ISSUE TITLE
Series Map does not have implementation for dask Series

### ISSUE THREAD

**Issue by bluecoconut** [dask/dask#4870]
Map Series does not work for dask Series mapped to dask Series. 

## Example
```
from dask.datasets import timeseries
df = timeseries()
xsum_per_name = df.groupby('name')['x'].sum(split_out=5).persist()
out = df['name'].map(xsum_per_name)
```

Get Error: 
```TypeError: arg must be pandas.Series, dict or callable. Got <class 'dask.dataframe.core.Series'>```

**Comment by bluecoconut** [comment]
Included a PR to address: https://github.com/dask/dask/pull/4872

**Comment by jakirkham** [comment]
Thanks for raising and submitting a fix, @bluecoconut. 😄

**Comment by jrbourbeau** [comment]
Closed by #4872
